src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/DefaultComesLastCheck.java

```diff
@@ -127,25 +127,20 @@ public void setSkipIfLastAndSharedWithCase(boolean newValue) {
     @Override
     public void visitToken(DetailAST ast) {
         final DetailAST defaultGroupAST = ast.getParent();
-        //default keywords used in annotations too - not what we're
-        //interested in
-        if (defaultGroupAST.getType() != TokenTypes.ANNOTATION_FIELD_DEF
-                && defaultGroupAST.getType() != TokenTypes.MODIFIERS) {
```

Comment:
`default` in these areas can't have a `CASE_GROUP` sibling.

Reply:
It was an optimization for `default` in annotations,  to avoid a call to `findNextSibling`.
But this optimization is questionable. I'm OK with this change.